app/controllers/sessions_controller.rb

```diff
@@ -5,19 +5,51 @@ class SessionsController < Clearance::SessionsController
   def create
     @user = find_user
 
-    if @user&.mfa_enabled?
-      session[:mfa_user] = @user.display_id
-      render "sessions/otp_prompt"
+    if @user && (@user.mfa_enabled? || @user.webauthn_credentials.any?)
+      setup_webauthn_authentication
+      setup_mfa_authentication
+
+      render "sessions/prompt"
     else
       do_login
     end
   end
 
+  def webauthn_create
+    @user = User.find_by_slug!(session.dig(:webauthn_authentication, "user"))
```

Comment:
This probably needs to change to id re: https://github.com/rubygems/rubygems.org/commit/86219b13079ad87e3d56adb07ea8899d42a39edb

Reply:
Resolved in https://github.com/rubygems/rubygems.org/pull/2865/commits/d0cd602ce3432e910bf49bc710e4eddd007b1dd4